shipyard_blueprints/sqlserver/shipyard_sqlserver/sqlserver.py

```diff
@@ -38,7 +34,7 @@ def __init__(
     @property
     def conn(self):
         if self._conn is None:
-            self.connect()
+            self._conn = self.connect()
         return self._conn
 
     def connect(self):
```

Comment:
not sure how much this matters but most connects return an int. This helps by not needing a separate  try clause in the auth test.

Reply:
I'm actually not a fan of the `connect` returning an int, especially in the database context. In typical database API's, a connect method returns the actual connection object. I prefer sticking to that paradigm and formulating the auth test around that, versus formulating the `connect` method around the auth test.